dcm4che-conf/dcm4che-conf-dicom/src/test/java/org/dcm4che3/conf/dicom/DicomConfigurationTest.java

```diff
@@ -91,6 +93,88 @@ public void renameAETest() throws ConfigurationException {
 
     }
 
+    @Test
+    public void listDevices_produce_expected_results() {
+        CommonDicomConfiguration config = SimpleStorageTest.createCommonDicomConfiguration();
+        config.purgeConfiguration();
+
+        Device originalDevice = createDeviceWithArchiveExtension("DEVICE", "Default", "Other", "Storage", DeviceType.ARCHIVE, DeviceType.PRINT);
+        config.persist(originalDevice);
+
+        Device[] devices = config.listDevices(DeviceType.ARCHIVE);
+        Assert.assertTrue(devices.length == 1);
+
+        Device device = devices[0];
+
+        Assert.assertTrue(deepCompare(originalDevice, device));
```

Comment:
I would prefer you add a message to each `Assert.assertTrue` statement about what failed. Otherwise, if there is a test failure, then the message returned will be, "True is False". So for each of these statements you should include something like, "Original device does not match the device returned by listDevices".